src/command/which.rs

```diff
@@ -17,14 +18,42 @@ pub(crate) struct Which {
 }
 
 impl Command for Which {
+    // 1. Start by checking if the user has a tool installed in the project or
+    //    as a user default. If so, we're done.
+    // 2. Otherwise, use the platform image and/or the system environment to
+    //    determine a lookup path to run `which` in.
     fn run(self, session: &mut Session) -> Fallible<ExitCode> {
         session.add_event_start(ActivityKind::Which);
 
+        let bin = OsString::from(self.binary.as_str());
+
+        let user_tool = session.get_user_tool(&bin)?;
+        let project_bin_path =
+            session
+                .project()?
+                .and_then(|project| match project.has_direct_bin(&bin) {
+                    Ok(true) => Some(project.local_bin_dir()),
+                    _ => None,
+                });
+
+        let tool_path = match (user_tool, project_bin_path) {
+            (Some(_), Some(project_path)) => Some(project_path),
```

Comment:
This looks like it will only show the directory that the project local bin is in (i.e. `.../node_modules/.bin`), but won't include the actual file binary file like the other returns from `which` do.

Reply:
Interesting; this actually is simply failing to resolve correctly, and I'll need to dig further to figure out why.

Reply:
It looks like it's working for me, apart from needing to `push` `&bin` to the end of the Path for the project local one (I locally updated `voyager-web` to use `volta` instead of `toolchain`):

```bash
$ volta which node
/Users/cpierce/.volta/tools/image/node/10.15.3/6.4.1/bin/node
$ volta which yarn
/Users/cpierce/.volta/tools/image/yarn/1.16.0/bin/yarn
$ volta which ember
/Users/cpierce/.volta/tools/image/packages/ember-cli/3.9.0/bin/ember
$ cd voyager-web
$ volta which node
/Users/cpierce/.volta/tools/image/node/10.6.0/6.1.0/bin/node
$ volta which yarn
/Users/cpierce/.volta/tools/image/yarn/1.13.0/bin/yarn
$ volta which ember
/Users/cpierce/code/voyager-web_trunk/node_modules/.bin
```

So all that's missing is adding `/ember` to the end of the path in the last case.

Reply:
Yeah, I just worked through the same; doing `.join(&bin)` resolves it nicely. Pushing that up now!

Reply:
Resolved in ba5cea4f!